internal/verify/verifier.go

```diff
@@ -0,0 +1,49 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fluxcd/source-controller/internal/verify/cosign"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/sigstore/cosign/pkg/oci"
+)
+
+// Enforce that Cosign implements Verifier interface.
+var _ Verifier = (*cosign.Verifier)(nil)
```

Comment:
I would expect this to be verified by the implementation, and not by the consumer.

Reply:
Additional note: better done in `_test.go` files, or not at all as Go build itself will complain if an implementation does not meet the needs of the contract.

Reply:
I've updated the code according to your reviews.

Reply:
PTAL @Dentrax